hal/halx86/acpi/acpitimer.c

```diff
@@ -97,85 +101,86 @@ NTAPI
 HalpPmTimerStallExecProc(_In_ ULONG MicroSeconds)
 {
     INT CpuInfo[4];
-    ULONGLONG Start;
-    ULONG StallValue;
-    LARGE_INTEGER Current;
+    ULONGLONG StartTick;
+    ULONG StallTicks;
+    LARGE_INTEGER TickCount;
     LARGE_INTEGER Next;
     ULONG TotalStall;
     ULONG StallCounter;
 
     //DPRINT1("HalpPmTimerStallExecProc: MicroSeconds %X\n", MicroSeconds);
 
-    /* Serializing instruction execution */
-    __cpuid(CpuInfo, 0);
-
     if (!MicroSeconds)
     {
         goto Exit;
     }
 
-    //HalpStallExecCounter++; // Statistics
+    /* Serializing instruction execution */
+    __cpuid(CpuInfo, 0);
 
-    StallValue = (PMTimerFreq * (ULONGLONG)MicroSeconds) / 1000000 - 1; // ~2,579545 * MicroSeconds
-    Start = QueryTimer();
+    /* Calculate the number of ticks for a given delay */
+    StallTicks = (PMTimerFreq * (ULONGLONG)MicroSeconds) / 1000000 - 1;
+
+    /* Reading the starting value of ticks */
+    StartTick = QueryTimer();
 
     TotalStall = 0;
     StallCounter = StallLoopValue;
 
     while (TRUE)
     {
+        /* Do one microsecond stall */
         while (TRUE)
         {
-            TotalStall += StallCounter;
-
-            _mm_pause();
```

Comment:
YieldProcessor()

Reply:
This is changed after the function refactoring.